fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -3959,6 +3982,35 @@ public void modifyTableEnablePersistentIndexMeta(Database db, OlapTable table, M
 
     }
 
+    public void modifyBinlogMeta(Database db, OlapTable table, BinlogConfig binlogConfig) {
+        Preconditions.checkArgument(db.isWriteLockHeldByCurrentThread());
+        binlogConfig.incVersion();
+        table.setCurBinlogConfig(binlogConfig);
+
+        Map<Long, Long> curAllTableIdsInBinlogManager = null;
+        if (table.enableBinlog()) {
+            Long nextTxnId = GlobalStateMgr.getCurrentGlobalTransactionMgr().getTransactionIDGenerator().getNextTransactionId();
+            BinlogManager binlogManager = GlobalStateMgr.getCurrentState().getBinlogManager();
+            try {
+                Set<Long> transactionIds = GlobalStateMgr.getCurrentGlobalTransactionMgr().
+                        getDatabaseTransactionMgr(db.getId()).getRunningTransactionsId();
+                binlogManager.recordBinlogTxnId(nextTxnId, db.getId(), table.getId(), transactionIds);
+                curAllTableIdsInBinlogManager = binlogManager.getAllTableIds();
+            } catch (AnalysisException e) {
+                LOG.warn(e.getMessage());
+            }
+
+        } else {
+            for (Partition partition : table.getAllPartitions()) {
```

Comment:
No need to set `-1` if just ttl or max size is modified

Reply:
Even if only modify ttl or max_size  setting it to -1 is ok。 The reason for this is that when binliogAvailableVerison is not available, there is no need to write editlog